tests/functional/src/main/java/org/mule/functional/classloading/isolation/classpath/DefaultMavenArtifactToClassPathURLResolver.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.functional.classloading.isolation.classpath;
+
+import org.mule.functional.classloading.isolation.maven.MavenArtifact;
+import org.mule.functional.classloading.isolation.maven.MavenArtifactToClassPathURLResolver;
+import org.mule.functional.classloading.isolation.maven.MavenMultiModuleArtifactMapping;
+
+import java.io.File;
+import java.net.URL;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Default implementation of {@link MavenArtifactToClassPathURLResolver} that supports artifacts already packages (for CI environments)
+ * and multi-module maven projects.
+ *
+ * @since 4.0
+ */
+public class DefaultMavenArtifactToClassPathURLResolver implements MavenArtifactToClassPathURLResolver
```

Comment:
I don't see why there are interface/default implementation everywhere, if there is no other usage of the interface than in the default, there is no factory for them, is not used on mocks and it does not looks like never there will be a different implementation other than the default one.